support/k8s/k8s-workload-registrar/mode-crd/webhook/webhook_entry.go

```diff
@@ -0,0 +1,251 @@
+// For example if the k8s-workload-registrar was force deleted and did not get
+// a chance to delete its webhook entry
+func (e *Entry) CleanupStaleEntries() error {
+	list, err := e.getWebhookEntries()
+	if err != nil {
+		return err
+	}
+
+	for _, entry := range list.Entries {
+		namespacedName, uid := selectorsToNamespacedNameAndUID(entry.Selectors)
+		if namespacedName == nil || (namespacedName.Name == e.c.Name &&
+			namespacedName.Namespace == e.c.Namespace &&
+			uid == e.c.UID) {
+			// Skip lookup for entry beloging to this pod
+			continue
+		}
+		pod, err := e.getPod(namespacedName)
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				// Resource has been deleted
+				if err = e.deleteEntry(entry.Id); err != nil {
+					return err
+				}
+				continue
+			}
+			return err
+		}
+		if pod.UID != uid {
+			if err = e.deleteEntry(entry.Id); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// DeleteEntry deletes the registration entry for the webhook
+func (e *Entry) DeleteEntry() error {
+	if e.entryID != "" {
+		if err := e.deleteEntry(e.entryID); err != nil {
+			return err
+		}
+	}
+	e.entryID = ""
+	e.SpiffeID = nil
+	return nil
+}
+
+// deleteEntry deletes an entry by ID
+func (e *Entry) deleteEntry(entryID string) error {
+	resp, err := e.c.E.BatchDeleteEntry(e.c.Ctx, &entryv1.BatchDeleteEntryRequest{Ids: []string{entryID}})
+	if err != nil {
+		return err
+	}
+
+	// These checks are purely defensive
+	switch {
+	case len(resp.Results) > 1:
+		return errors.New("batch delete response has too many results")
+	case len(resp.Results) < 1:
+		return errors.New("batch delete response missing result")
+	}
+
+	err = errorFromStatus(resp.Results[0].Status)
+	switch status.Code(err) {
+	case codes.OK, codes.NotFound:
+		return nil
+	default:
+		return err
+	}
+}
+
+// webhookParentID creates the SPIFFEID for the PSAT parent used by the webhook
+func (e *Entry) webhookParentID() (*spiretypes.SPIFFEID, error) {
+	nodeUID, err := e.nodeNameToUID()
+	if err != nil {
+		return nil, err
+	}
+	return &spiretypes.SPIFFEID{
+		TrustDomain: e.c.TrustDomain,
+		Path:        fmt.Sprintf("spire/agent/k8s_psat/%s/%s", e.c.Cluster, nodeUID),
+	}, nil
+}
+
+// webhookSpiffeID creates the SPIFFEID for the webhook
+func (e *Entry) webhookSpiffeID() *spiretypes.SPIFFEID {
+	return &spiretypes.SPIFFEID{
+		TrustDomain: e.c.TrustDomain,
+		Path:        fmt.Sprintf("/k8s-workload-registrar/%s/webhook", e.c.Cluster),
+	}
+}
+
+// nodeNameToUID converts an node name to a corresponding UID by calling the Kubernetes API
+func (e *Entry) nodeNameToUID() (types.UID, error) {
+	node, err := e.c.Clientset.CoreV1().Nodes().Get(e.c.Ctx, e.c.NodeName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	return node.UID, nil
+}
+
+// getWebhookEntries gets a list of the all the entries for the webhook
+func (e *Entry) getWebhookEntries() (*entryv1.ListEntriesResponse, error) {
+	return e.c.E.ListEntries(e.c.Ctx, &entryv1.ListEntriesRequest{
```

Comment:
It will if the annotation/label entry has the same spiffe id. The entry won't be deleted unless the pod its attached to has been deleted. It checks to make sure no pod with a uid matching the selector uid exists before deleting it. In that case it would be deleted anyways, so it won't cause a problem. https://github.com/spiffe/spire/pull/2321/files#diff-9355d9bfd9c454f8c9630d5a02c2c50dcf14fd6c5f23b1175021b5a009ac3100R124-R139